libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -1464,6 +1530,27 @@ export class RushConfiguration {
     return this._projectsByTag;
   }
 
+  /**
+   * Obtains the mapping from splitWorkspace to projects.
+   * @beta
+   */
+  public get projectsBySplitWorkspace(): ReadonlyMap<boolean, ReadonlySet<RushConfigurationProject>> {
```

Comment:
A map is an unusual choice for this data structure, since there are only two possible keys, and the meaning of `true` vs `false` is unclear.  Also storing `RushConfigurationProject` in a set is slightly add, since the equality is based on object instances. (Is it possible to somehow obtain two different `RushConfigurationProject` objects that represent the same project name and file path? An API user would consider them to be the "same" project, but the `Set<>` considers them to be different.)

Compare with a simple design like:
```ts
  rushConfiguration.projects.filter(x =>x.splitWorkspace)
```
When enumerating all projects, the time complexity of `O(all projects)` is not that much more than `O(filtered projects)` and maybe not worth optimizing at all.

But an operation like `hasSplitWorkspaceProject()` could possibly benefit from caching.  If we want to optimize that, it could be a simpler design like:

```ts
  rushConfiguration.getFilteredProjects({ splitWorkspace: false });
```
maybe implemented like:
```ts
  interface IRushConfigurationProjectsFilter { splitWorkspace?: boolean }

  public getFilteredProjects(filter: IRushConfigurationProjectsFilter): RushConfigurationProject[] {
    const cacheKey: string = `${filter.splitWorkspace}`;

    let result: RushConfigurationProject[] | undefined = this._filteredProjectsCache.get(cacheKey);
    if (!result) {
      result = this.projects.filter(x => filter.splitWorkspace === undefined || x.splitWorkspace === filter.splitWorkspace);
      this._filteredProjectsCache.set(cacheKey, result);
    }
    return result;
  }
```